src/crdt-helpers.ts

```diff
@@ -53,6 +53,35 @@ function toString<K extends IndexKeyType>(key: K, logger: Logger): string {
   }
 }
 
+export function sanitizeDocumentFields(obj: unknown): unknown {
+  if (Array.isArray(obj)) {
+    return obj.map((item: unknown) => {
+      if (typeof item === 'object' && item !== null) {
+        return sanitizeDocumentFields(item);
+      }
+      return item;
+    });
+  } else if (typeof obj === 'object' && obj !== null) {
+    const typedObj = obj as Record<string, unknown>;
```

Comment:
this is also entered if it is a class instances

Reply:
I think that should be OK, we just want to clean out undefined fields